push-get-payload/server.js

```diff
@@ -2,25 +2,43 @@
 // between the application server and the push service.
 // For details, see https://tools.ietf.org/html/draft-ietf-webpush-protocol and
 // https://tools.ietf.org/html/draft-ietf-webpush-encryption.
-var webPush = require('web-push');
+const webPush = require('web-push');
 
-webPush.setGCMAPIKey(process.env.GCM_API_KEY || null);
+if (!process.env.VAPID_PUBLIC_KEY || !process.env.VAPID_PRIVATE_KEY) {
+  console.log("You must set the VAPID_PUBLIC_KEY and VAPID_PRIVATE_KEY "+
+    "environment variables. You can use the following ones:");
+  console.log(webPush.generateVAPIDKeys());
+  return;
+}
+// Set the keys used for encrypting the push messages.
+webPush.setVapidDetails(
+  'https://serviceworke.rs/',
+  process.env.VAPID_PUBLIC_KEY,
+  process.env.VAPID_PRIVATE_KEY,
+);
 
-var payloads = {};
+const payloads = {};
 
 module.exports = function(app, route) {
+  app.get(route + 'vapidPublicKey', function(req, res) {
+    res.send(process.env.VAPID_PUBLIC_KEY);
+  });
+
   app.post(route + 'register', function(req, res) {
     // A real world application would store the subscription info.
     res.sendStatus(201);
   });
 
   app.post(route + 'sendNotification', function(req, res) {
+    const subscription = req.body.subscription;
+    const payload = req.body.payload;
+    const options = {
+      TTL: req.body.ttl
+    };
+
     setTimeout(function() {
-      payloads[req.body.endpoint] = req.body.payload;
-      webPush.sendNotification({
-        endpoint: req.body.endpoint,
-        TTL: req.body.ttl,
-      })
+      payloads[req.body.endpoint] = payload;
```

Comment:
You need to index the `payloads` map by the endpoint which is now [*inside* the `subscription` object](https://github.com/mozilla/serviceworker-cookbook/pull/294/files#diff-12d95901e9b60d3002f4388bbc702b06R52): 
```js
payloads[req.body.subscription.endpoint] = payload;
```

Reply:
Oops! Strange I didn't notice this. Fixed.